### PR TITLE
[Part 1] Implement windowsbrush

### PR DESCRIPTION
**Pull request by marlenecota**
This commit adds the logic that handles receiving an object for a color value and mapping it to a XAML theme brush or system accent color. 

Part 2 requires changes to our react-native fork (and eventually fb) or adding a .uwp.js implementation for stylesheet and any other component that has a color prop. 

In the meantime, if you'd like to locally see these changes in action, make the following changes:
1. node_modules\react-native\Libraries\Color\normalizeColor.js
```Javascript
// Line 41 - Add Platform.OS === 'uwp'
if (typeof color === 'object' && color !== null && (Platform.OS === 'macos' || Platform.OS === 'uwp')) { 
    // Add 'windowsbrush' in color
    if ('semantic' in color || 'windowsbrush' in color) { 
      return color;
    } else if ('dynamic' in color && color.dynamic !== undefined) {
```
2. node_modules\react-native\Libraries\StyleSheet\processColor.js
```Javascript
if (typeof int32Color === 'object' &&
    (Platform.OS === 'macos' || Platform.OS === 'uwp') // Line 33 - Add Platform.OS === 'uwp'
  ) {
    if ('dynamic' in int32Color && int32Color.dynamic !== undefined) {
```
Sample usage:
```JSX
<Button title='System Accent Color' color={{windowsbrush: 'SystemAccentColorLight3'}}></Button>
<Button title='XAML Theme Brush' color={{windowsbrush: 'SystemControlForegroundAccentBrush'}}></Button>
````

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2705)

### CODE REVIEW

**Comment by licanhua** [comment]
Can `<Button title='System Accent Color' color={{windowsbrush: 'SystemAccentColorLight3'}}></Button>`
be
`<Button title='System Accent Color' color={windowsbrush: 'SystemAccentColorLight3'}></Button>`

File: vnext/ReactUWP/Utils/ValueUtils.cpp [hunk 1]
```diff
@@ -49,13 +50,51 @@ REACTWINDOWS_API_(winrt::Color) ColorFrom(const folly::dynamic& d)
 
 REACTWINDOWS_API_(winrt::SolidColorBrush) SolidColorBrushFrom(const folly::dynamic& d)
 {
-  thread_local static std::map<winrt::Color, winrt::weak_ref<winrt::SolidColorBrush>, ColorComp> solidColorBrushCache;
+  if (d.isObject())
+  {
+    winrt::hstring resourceName{ winrt::to_hstring(d.find("windowsbrush")->second.asString()) };
+
+    thread_local static std::map<winrt::hstring, winrt::weak_ref<winrt::SolidColorBrush>> accentColorMap = {
```
Comment (licanhua) [inline vnext/ReactUWP/Utils/ValueUtils.cpp]: >accentColorMap [](start = 90, length = 14)

Do you need to refresh the map when theme is changed?

Comment (marlenecota) [reply vnext/ReactUWP/Utils/ValueUtils.cpp]: No, the ThemeResource keyword takes care of that.

**Comment by marlenecota** [comment]
No, it's like with inline styles style={{color: 'blue'}}. You only need one set of brackets in a StyleSheet though:

<View style={styles.accent}/>

...

const styles = StyleSheet.create({
  accent: {
    backgroundColor: {windowsbrush: 'SystemAccentColor'}
  }
});

---
In reply to: [507413807](https://github.com/microsoft/react-native-windows/pull/2705#issuecomment-507413807) [](ancestors = 507413807)

File: vnext/ReactUWP/Utils/ValueUtils.cpp [hunk 2]
```diff
@@ -49,13 +50,51 @@ REACTWINDOWS_API_(winrt::Color) ColorFrom(const folly::dynamic& d)
...
+      {
+        return brush;
+      }
+
+      winrt::hstring xamlString =
+        L"<ResourceDictionary xmlns = 'http://schemas.microsoft.com/winfx/2006/xaml/presentation' xmlns:x = 'http://schemas.microsoft.com/winfx/2006/xaml'>"
+        L"  <SolidColorBrush x:Key='" + resourceName + L"' Color='{ThemeResource " + resourceName + "}' />"
+        L"</ResourceDictionary>";
+
+      auto dictionary{ winrt::unbox_value<winrt::ResourceDictionary>(winrt::Markup::XamlReader::Load(xamlString)) };
```
Comment (kmelmon) [inline vnext/ReactUWP/Utils/ValueUtils.cpp]: @jonthysell does XamlDirect support creating ThemeResources?  If so I think I'd prefer that to XamlReader.Load()

File: vnext/ReactUWP/Utils/ValueUtils.cpp [hunk 3]
```diff
@@ -49,13 +50,51 @@ REACTWINDOWS_API_(winrt::Color) ColorFrom(const folly::dynamic& d)
...
+      auto brush{ winrt::unbox_value<winrt::SolidColorBrush>(dictionary.Lookup(winrt::box_value(resourceName))) };
+      accentColorMap[resourceName] = winrt::make_weak(brush);
+      return brush;
+    }
+
+    winrt::IInspectable resource{ winrt::Application::Current().Resources().Lookup(winrt::box_value(resourceName)) };
+    return winrt::unbox_value<winrt::SolidColorBrush>(resource);
```
Comment (kmelmon) [inline vnext/ReactUWP/Utils/ValueUtils.cpp]: >return winrt::unbox_value<winrt::SolidColorBrush>(resource); [](start = 4, length = 60)

If the app supplies a name that isn't recognized, will we crash?  Should we fallback to a transparent brush instead (or some other fallback)?

**Pull request merged**